adapters/improvedigital/improvedigital.go

```diff
@@ -36,11 +57,16 @@ func (a *ImprovedigitalAdapter) MakeRequests(request *openrtb2.BidRequest, reqIn
 
 func (a *ImprovedigitalAdapter) makeRequest(request openrtb2.BidRequest, imp openrtb2.Imp) (*adapters.RequestData, error) {
 	request.Imp = []openrtb2.Imp{imp}
+
 	reqJSON, err := json.Marshal(request)
 	if err != nil {
 		return nil, err
 	}
 
+	if reqCloneJSON, err := a.getJSONWithAdditionalConsent(request, reqJSON); err == nil {
```

Comment:
This feels wrong. You should be manipulating a clone of `request`, and not its JSON form, having to unmarshal it first then marshal it again. 
So please look into how to simply add additional field consented_providers.consented_providers_settings into a cloned `request`, don't touch other fields inside `user.ext`  including `ConsentedProvidersSettings`. Just add the new field `consented_providers`. `user.ext` can in the future contain more fields than you listed in `UserExtReq` and with your current implementation we would lose any new field.